File: reverseInteger.js
/*
Given a 32-bit signed integer, reverse digits of an integer.

Note:
Assume we are dealing with an environment that could only store integers within the
32-bit signed integer range: [−231,  231 − 1]. For the purpose of this problem, assume that your
function returns 0 when the reversed integer overflows.

Example 1:
Input: x = 123
Output: 321

Example 2:
Input: x = -123
Output: -321

Example 3:
Input: x = 120
Output: 21

Example 4:
Input: x = 0
Output: 0
*/

var reverse = function(x) {
    let isNegative = x < 0 ? true: false;
    let result = 0;
    
    if (isNegative) {
        x = x * (-1);
    }
    
    while (x !== 0) {
        result = result * 10 + x % 10;
        x = Math.floor(x / 10);
    }
    
    if (result >= 2147483648) {
        return 0;
    }
    
    return isNegative ? result * (-1) : result;
};

File: Python/reverseInteger.py
"""
Given a signed 32-bit integer x, return x with its digits reversed. If reversing x causes the value to go outside the signed 32-bit integer range [-231, 231 - 1], then return 0.

Assume the environment does not allow you to store 64-bit integers (signed or unsigned).

Example 1:
Input: x = 123
Output: 321

Example 2:
Input: x = -123
Output: -321

Example 3:
Input: x = 120
Output: 21

Example 4:
Input: x = 0
Output: 0

https://leetcode.com/problems/reverse-integer/
"""

import math
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        factor = -1 if x < 0 else 1
        reversed_number = 0
        while x != 0:
            reversed_number = (reversed_number * 10) + math.floor(x % 10)
            x = math.floor(x//10)
            
        if reversed_number >= 2147483648:
            return 0
        
        return int(factor * reversed_number)